From 043630dae96d6bf33abe64960c3917be57cfd296 Mon Sep 17 00:00:00 2001
From: Lily Zhang <r58066@freescale.com>
Date: Thu, 7 May 2009 09:08:58 +0800
Subject: [PATCH] ENGR00112237 Add FM support in MX51 28 kernel

1. Enable FM in default config
2. Fix GPIO request/free issue

Signed-off-by: Lily Zhang <r58066@freescale.com>
---
 arch/arm/configs/imx51_defconfig |    6 ++++--
 arch/arm/mach-mx51/mx51_3stack.c |    2 --
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/arm/configs/imx51_defconfig b/arch/arm/configs/imx51_defconfig
index ab85b1f..05d3cca 100644
--- a/arch/arm/configs/imx51_defconfig
+++ b/arch/arm/configs/imx51_defconfig
@@ -1,7 +1,7 @@
 #
 # Automatically generated make config: don't edit
 # Linux kernel version: 2.6.28
-# Wed Apr 29 10:11:33 2009
+# Thu May  7 08:51:48 2009
 #
 CONFIG_ARM=y
 CONFIG_SYS_SUPPORTS_APM_EMULATION=y
@@ -160,6 +160,7 @@ CONFIG_ARCH_MXC=y
 # CONFIG_ARCH_DAVINCI is not set
 # CONFIG_ARCH_OMAP is not set
 # CONFIG_ARCH_MSM is not set
+# CONFIG_ARCH_STMP3XXX is not set
 
 #
 # Boot options
@@ -739,7 +740,7 @@ CONFIG_HW_CONSOLE=y
 # CONFIG_VT_HW_CONSOLE_BINDING is not set
 CONFIG_DEVKMEM=y
 # CONFIG_SERIAL_NONSTANDARD is not set
-# CONFIG_FM_SI4702 is not set
+CONFIG_FM_SI4702=m
 CONFIG_MXC_IIM=y
 
 #
@@ -1827,6 +1828,7 @@ CONFIG_CRYPTO=y
 # CONFIG_CRYPTO_CRYPTD is not set
 # CONFIG_CRYPTO_AUTHENC is not set
 # CONFIG_CRYPTO_TEST is not set
+# CONFIG_CRYPTO_CRYPTODEV is not set
 
 #
 # Authenticated Encryption with Associated Data
diff --git a/arch/arm/mach-mx51/mx51_3stack.c b/arch/arm/mach-mx51/mx51_3stack.c
index ac53348..27dd653 100644
--- a/arch/arm/mach-mx51/mx51_3stack.c
+++ b/arch/arm/mach-mx51/mx51_3stack.c
@@ -330,13 +330,11 @@ void si4702_clock_ctl(int flag)
 static void si4702_gpio_get(void)
 {
 	/* reset pin */
-	mxc_request_iomux(MX51_PIN_EIM_DTACK, IOMUX_CONFIG_GPIO);
 	mxc_set_gpio_direction(MX51_PIN_EIM_DTACK, 0);
 }
 
 static void si4702_gpio_put(void)
 {
-	mxc_free_iomux(MX51_PIN_EIM_DTACK, IOMUX_CONFIG_GPIO);
 }
 
 static struct mxc_fm_platform_data si4702_data = {
-- 
1.5.4.4

